src/nrc/api/tests/test_scopes.py

```diff
@@ -0,0 +1,344 @@
+from rest_framework.test import APITestCase
+from vng_api_common.tests import JWTAuthMixin, get_operation_url, reverse
+
+from nrc.api.choices import SequencetypeChoices
+from nrc.datamodel.choices import ProtocolChoices
+from nrc.datamodel.tests.factories import DomainFactory, SubscriptionFactory
+
+from ..scopes import (
+    SCOPE_DOMAINS_CREATE,
+    SCOPE_DOMAINS_READ,
+    SCOPE_EVENTS_PUBLISH,
+    SCOPE_SUBSCRIPTIONS_CREATE,
+    SCOPE_SUBSCRIPTIONS_DELETE,
+    SCOPE_SUBSCRIPTIONS_READ,
+    SCOPE_SUBSCRIPTIONS_UPDATE,
+)
+
+
+@override_settings(
+    LINK_FETCHER="vng_api_common.mocks.link_fetcher_200",
+    ZDS_CLIENT_CLASS="vng_api_common.mocks.MockClient",
+)
+class ScopeDomains(JWTAuthMixin, APITestCase):
+    scopes = []
+
+    def test_correct_scope_post(self):
+        """
+        test /domains POST:
+        create domain with correct scope SCOPE_DOMAINS_CREATE
+        """
+        self.autorisatie.scopes = [SCOPE_DOMAINS_CREATE]
+        self.autorisatie.save()
+        data = {"name": "zaken", "documentation_link": "https://example.com/doc"}
+
+        domain_create_url = get_operation_url("domain_create")
+        response = self.client.post(domain_create_url, data)
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, response.data)
+
+    def test_incorrect_scope_post(self):
+        """
+        test /domains GET:
+        read domain with incorrect scope SCOPE_DOMAINS_READ
+        """
+        self.autorisatie.scopes = [SCOPE_DOMAINS_READ]
+        self.autorisatie.save()
+        data = {"name": "zaken", "documentation_link": "https://example.com/doc"}
+
+        domain_create_url = get_operation_url("domain_create")
+        response = self.client.post(domain_create_url, data)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_correct_scope_list(self):
+        """
+        test /domains GET:
+        retrieve domain with correct scope SCOPE_DOMAINS_READ
+        """
+        self.autorisatie.scopes = [SCOPE_DOMAINS_READ]
+        self.autorisatie.save()
+        domain = DomainFactory()
+
+        url = get_operation_url("domain_read", uuid=domain.uuid)
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_incorrect_scope_list(self):
+        """
+        test /domains GET:
+        retrieve domain with incorrect scope SCOPE_DOMAINS_CREATE
+        """
+        self.autorisatie.scopes = [SCOPE_DOMAINS_CREATE]
+        self.autorisatie.save()
+        domain = DomainFactory()
+
+        url = get_operation_url("domain_read", uuid=domain.uuid)
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN, response.data)
+
+
+@override_settings(
+    LINK_FETCHER="vng_api_common.mocks.link_fetcher_200",
+    ZDS_CLIENT_CLASS="vng_api_common.mocks.MockClient",
+)
+class ScopeSubscriptions(JWTAuthMixin, APITestCase):
```

Comment:
Probeer ook consistente naamgeving te gebruiken in de namen van je tests, ik zie bijvoorbeeld `...._put` voorbij komen maar ook `...._update`. Een van de twee zou ik overal consistent aanhouden.

Reply:
ja goed punt, dat was een grote rommel. Ik pas het aan naar de namen van de scopre.